src/TraceStream.cc

```diff
@@ -592,6 +599,14 @@ TraceReader::TraceReader(const string& dir)
   }
 
   string path = version_path();
+  if (!file_exists(path)) {
+    fprintf(
+        stderr,
+        "rr: warning: No traces have been recorded so far.\n"
```

Comment:
This message might not be true. It might just be that someone did `rr replay <path>` where `<path>` is not the name of an rr trace directory.

Also, I think we don't need a separate `file_exists` check here. If `vfile.good()` is false then we can assume the file doesn't exist. We should just print a different, better message saying that the trace was not found.

Actually what would be really useful is to change this constructor so that if an error occurs, it returns an error code (an enum) in an out-parameter and the caller can handle it. Then `ReplaySession`'s constructor can forward that error out to its caller, `ReplayCommand`, which can display the error message. Really the Command classes should be responsible for interacting with the user.